### ISSUE TITLE
Try-With-Resource broken with Java 8

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext-extras#445]
when running `org.eclipse.xtext.purexbase.test.XbaseIntegrationTest`
against java 8 target (we use 5 as default) it will fail
https://ci.eclipse.org/xtext/job/xtext-extras/job/cd_extras_issue113/lastCompletedBuild/testReport/org.eclipse.xtext.purexbase.test/XbaseIntegrationTest/testTryWithResources_finally/

the same problem can be reproduce with a simple xtend sample

```
package org.eclipse.xtext.purexbase.test

class Demo {

	def xxxx() {
		var testdata.ClosableWithList result
		try (var a = new testdata.ClosableWithList()) {
			a.add("body")
			result = a
		} finally {
			a.add("finally")
		}
		return result.printList
	}
}

```

```
public class Demo {
  public String xxxx() {
    ClosableWithList result = null;
    try (ClosableWithList a = new ClosableWithList()) {
      a.add("body");
      result = a;
    } finally {
      a.add("finally");
    }
    return result.printList();
  }
}
```

a cannot be resolved

**Comment by cdietrich** [comment]
=> with a should not be in scope in finally or we should rethink compilation strategy
cc @szarnekow

**Comment by szarnekow** [comment]
`a` must not be reachable in the finally expression. I'll look at it.

**Comment by cdietrich** [comment]
maybe we should have a look at the problems i have with eclipse/xtext#2344 to avoid such things for the future

**Comment by szarnekow** [comment]
Fixed

**Comment by cdietrich** [comment]
Fixed in 2.19
